{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1e8ad43f-2db9-4bce-ab22-d8f3031831e6",
   "metadata": {},
   "source": [
    "## Geneious analysis for individual samples from raw Geneious output, \"Annotation.csv\" \n",
    "\n",
    " ### Required packages\n",
    " - No specific package required \n",
    " \n",
    " ### Inputs\n",
    " - Geneious SNP analysis of _k13_, _crt_, _mdr1_, _dhfr_, _dhps_, and _cytb_ for pooled samples\n",
    " - Documentation on Geneious analysis can be found: Readme.md\n",
    " - Geneious outputs were modified to GuineaAnalysis_pooled.csv from \"Annotation.csv\"\n",
    " \n",
    " \n",
    " ### Data structure \n",
    " - [Long-form](https://seaborn.pydata.org/tutorial/data_structure.html#long-form-vs-wide-form-data) \n",
    "     - Each variable is a column \n",
    "\n",
    "         - \"Sample\" = *AMD ID*, including associated meta-data for each sample\n",
    "             - AMD ID and bit code key is found under MS Teams > Domestic > Files > Sample Naming > Sample_naming_key.pptx  \n",
    "\n",
    "             - Key: **Year Country State/Site Treatment SampleID Genus SampleType GeneMarker-8bitcode SampleSeqCount**\n",
    "\n",
    "                 - Example:\n",
    "                     - Individual sequenced sample ID: 17GNDo00F0001PfF1290 = 2017 Guinea Dorota Day0 AS+AQ 0001 P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                     - Pooled sequenced sample ID: 17GNDoxxx001P10F1290 = 2017 Guinea Dorota **xx x** 001 **Pooled SamplesInPool** P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                         - NOTE: If information is not availble (na) **x** is used. For pooled samples, DayofTreatment and Treatment is na since its a pool of multiple samples with that info. \n",
    "                         - NOTE: For pooled samples, **Genus** is replaced with **Pooled** and **SampleType** with **SamplesInPool** to indicated this as a pooled sequenced sample and sample count in each pool. \n",
    "         <p>&nbsp;</p>\n",
    "         - \"Pooled\" = whether this sample is a pooled or an individual sample\n",
    "         - \"Year\" = the year the study was conducted \n",
    "         - \"Site\" = the state or province \n",
    "         - \"Gene\" = drug resistant gene(s) \n",
    "         - \"Coverage\" = the number of reads covering the SNP \n",
    "         - \"VAF\" = variant allele frequency calculated by AA divided by total reads in loci \n",
    "         - \"SNP\" = single nucleotide polymorphism in WildTypeAA or MutantAA annotation format \n",
    "         - \"Type\" = describes if it is a wild type or mutant SNP \n",
    "         - \"Pooled_size\" = describes how many individual samples were in the pool\n",
    "         - \"Pooled_group\" = describes which group the pooled sample belongs to \n",
    "         \n",
    "\n",
    "     - Each observation is a row for each sample ID (patient ID) \n",
    " \n",
    " #### TODO\n",
    " \n",
    " #### Activity Name\n",
    " - [ ] Write doc.string at the beginning of the code\n",
    " - [ ] Write detailed description with comment for line by line\n",
    " - [ ] Make the code more simple and accurate\n",
    " - [ ] Follow zen of python\n",
    "    \n",
    " #### Completed Activity âœ“\n",
    " - [x] Created marked down at the beginning of the file for description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c74c280a-d14b-42e0-bf95-3fa35f811400",
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "The purpose of this file is for representing the weighted average and sample sizes at the later stage so it will have less columns for the other pooled adding files\n",
    "\"\"\"\n",
    "\n",
    "###We open previously created Geneious analysis for both pooled and individual files#####\n",
    "with open (\"GuineaAnalysis.csv\", \"r\") as r1:   ##Open the previously created pooled lab file \n",
    "    ####We assign pooled group and pooled size to the previous information####\n",
    "    with open (\"GuineaAnalysis_ps.csv\", \"w\") as w1:  ##Open a new file because we would like to add pooled size and pooled group to the filess\n",
    "        ###We first write header for the new file####\n",
    "        w1.write(\"Sample,Pooled,SITE,GENE,SNP,COVERAGE,VAF,VF,PooledSize\\n\")  ##These are the columns \n",
    "        for lines in r1:\n",
    "            a=lines.split(\",\")[0]  ##This is the sample ID\n",
    "            test1=\"\"\n",
    "            ####For every line we add pooled size and pooled group####\n",
    "            ###This is for the pooled info part 1######\n",
    "            with open(\"Pooled_Info_Part1.csv\" , \"r\") as r2:  ##Check pooled part1 info to see group and size\n",
    "                for lines2 in r2:  ##Loop through pooled part 1 file\n",
    "                    if a in lines2:  ##Check if sample ID is in the given line to extract  the information about pooled size and pooled group\n",
    "                        if lines2[0:-3][-1].isdigit():  ##Check if the last number is digit. This was due to the nature of file was written that there was extra column at the end\n",
    "                            w1.write(lines.strip(\"\\n\")+\",\"+lines2[0:-3][-1]+\"\\n\")  ##If condition is met write extra two columns to the previous files\n",
    "                            test1=\"used\"  ##Assign test is used to check if the variable is used\n",
    "                        if lines2[0:-4][-1].isdigit():   ##Check if the last number is digit. This was due to the nature of file was written that there was extra column at the end\n",
    "                            w1.write(lines.strip(\"\\n\")+\",\"+lines2[0:-4][-1]+\"\\n\")  ##If condition is met write extra two columns to the previous files\n",
    "                            test1=\"used\"  ##Assign test is used to check if the variable is used\n",
    "            with open(\"Pooled_Info_Part2_fixed.csv\" , \"r\") as r3:  ##Also run a loop for pooled part 2\n",
    "                for lines3 in r3:  ##loop  through pooled part the file\n",
    "                    if a in lines3:  ##check if sample ID is in the file line\n",
    "                        #print(lines3)\n",
    "                        w1.write(lines.strip(\"\\n\")+\",\"+lines3.strip(\"\\n\")[-1]+\"\\n\")  ##If condition is met write extra two columns to the previous files\n",
    "                        test1=\"used\"  ##Assign test is used to check if the variable is used\n",
    "            if test1!=\"used\" and lines.startswith(\"Sample\")==False:  ##If none of the test is used just add pooled size of 1\n",
    "                w1.write(lines.strip(\"\\n\")+\",1\\n\")\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b63aca0-1c64-4f61-9b14-8276f23170b4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa4ea39f-10ab-49ea-a920-cfdf864da369",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
